addons/web/static/src/core/position_hook.js

```diff
@@ -98,10 +108,10 @@ function getBestPosition(reference, popper, { container, margin, position }) {
     // Compute positioning data
     /** @type {DirectionsData} */
     const directionsData = {
-        t: refBox.top - popBox.height - margin,
-        b: refBox.bottom + margin,
-        r: refBox.right + margin,
-        l: refBox.left - popBox.width - margin,
+        t: refBox.top + popMargins.top - popBox.height - margin,
+        b: refBox.bottom + popMargins.bottom + margin,
+        r: refBox.right + popMargins.right + margin,
+        l: refBox.left + popMargins.left - popBox.width - margin,
```

Comment:
Not sure what should be done, but I believe this is inconsistent (tell me if I'm wrong, I don't know this code in depth ^^):

if the desired intent is that the `popMargins` dimensions (applied style) should be enclosed in the container (along with the additional `margin` and the `popBox` dimensions), it should be:
```js
        t: refBox.top - popMargins.top - popBox.height - margin,
        b: refBox.bottom + popMargins.bottom + margin,
        r: refBox.right + popMargins.right + margin,
        l: refBox.left - popMargins.left - popBox.width - margin,
```
if instead we want to allow them to overlap the container (and only consider the additional `margin` as the actual margin of the popper) it should be:
```js
        t: refBox.top + popMargins.top - popBox.height - margin,
        b: refBox.bottom - popMargins.bottom + margin,
        r: refBox.right - popMargins.right + margin,
        l: refBox.left + popMargins.left - popBox.width - margin,
```

Reply:
Also, shouldn't we consider the margins twice ? -> in case all margins should be enclosed in the container it would become this:
```js
        t: refBox.top - popMargins.top - popMargins.bottom - popBox.height - 2 * margin,
        b: refBox.bottom + popMargins.bottom + popMargins.top + 2 * margin,
        r: refBox.right + popMargins.right + popMargins.left + 2 * margin,
        l: refBox.left - popMargins.left - popMargins.right - popBox.width - 2 * margin,
```

But tell me if I misunderstood something^^